### ISSUE TITLE
Add "by key" fields in GraphQL

### ISSUE THREAD

**Issue by rijkvanzanten** [directus/directus#4627]
Right now, the full CRUD operations can only be done on list-sets in GraphQL. We should have a way where you can operate on just a single item by key.

Adding a suffix to the query/mutation field name seems to be most convenient: `articles {...}` -> `articles_by_key(id: "abc") {...}`

(We _could_ alternatively use some way to guess the singular/plural name of a collection, by using something like https://graphql-compose.github.io/docs/api/misc-api-methods.html#getpluralname, but that feels super tricky cross-language)

**Comment by rijkvanzanten** [comment]
We should test this with Apollo's front-end caching systems to make sure it plays nicely with it

**Comment by iksent** [comment]
> We should test this with Apollo's front-end caching systems to make sure it plays nicely with it

I had a problems with Apollo's caching: https://github.com/directus/directus/discussions/4265#discussioncomment-482284
Switched to a [React Query](https://react-query.tanstack.com/) with another caching system - it works Ok.

**Comment by rijkvanzanten** [comment]
@iksent yeah! Seeing that Apollo is so popular though, it's worth looking into. (Super annoying that graphql tooling is so all over the place!)
